cpp/test/CMakeLists.txt

```diff
@@ -1,6 +1,9 @@
 # warning C4310: cast truncates constant value
 cxx_add_compile_options(MSVC /wd4310)
 
+# Core unit tests don't build under VS2013 without this.
+cxx_add_compile_options(MSVC /EHa)
```

Comment:
Let me see if I can figure out a way to eliminate this switch. It's a scary switch that make `catch(...)` also catch things like hardware exceptions.